protocol/src/main/kotlin/tbdex/sdk/protocol/Validator.kt

```diff
@@ -70,4 +74,31 @@ object Validator {
       throw ValidatorException(message = "invalid payload", errors = validationMessages.map { it.message })
     }
   }
+
+  /**
+   * Validate a message against a predefined schema
+   * @param message The message data to validate.
+   * @throws Exception if validation fails, including a list of validation errors.
+   */
+  fun validateMessage(message: Message) {
+    val mapper = ObjectMapper()
+    val messageJsonNode = mapper.readTree(message.toString())
+
+    validate(messageJsonNode, "message")
+    validateData(message.data, message.metadata.kind.toString())
+  }
+
+  /**
+   * Validate message data or resource data against a predefined schema
+   * @param data The message or resource data to validate.
+   * @param messageKind The message or resource kind of the data.
+   * @throws Exception if validation fails, including a list of validation errors.
+   */
+  fun validateData(data: Data, messageKind: String) {
+    val mapper = ObjectMapper()
```

Comment:
ditto: You can use the `jsonMapper` from `Json.kt` instead of reinstantiating

Reply:
done